### PR TITLE
several additions

### PR DESCRIPTION
**Pull request by VladimirAlexiev**
- reordered SHACL implementations by lang
- added maplib
- added ShEx Primers and Test suite
- small typographic changes

Tentatively assigning to @tpluscode. 
@amivanoff I'll also appreciate if you check it; unfortunately git doesn't let me assign  to you

### CODE REVIEW

**Comment by amivanoff** [comment]
I checked this PR with the linter, there are some errors.

1. It requires an awesome list item, formatted like this:

`<name with link, images and bages> <SPACE> <DASH> <SPACE> <description>`

So, to comply with the linter, we need to revert these dash-to-semicolon substitutions `" - "` => `": "`.

2. This change `Talks` => `Talks and Presentations` did not make it to the Contents.

3. The ShEx-validator misses `": "` or `" - "` separator. Also, the ShEx-validator was marked with one ☠️ as 6-year-old abandonware, according to the Legend section. Should we correct the Legend section or is there a more recent release?

4. Non-standard separators (like `", "`) in the Specifications/ShEx section.

**Comment by VladimirAlexiev** [comment]
> ShEx-validator was marked with one ☠️ as 6-year-old

You are right, I put the icon back, and also demoted Validata ☠️

> Non-standard separators (like ", ") 

Didn't quite get that but changed all commas to colons just in case :-)

Fixed all. @amivanoff could you please run the linter again?

**Comment by tpluscode** [comment]
> Fixed all. @amivanoff could you please run the linter again?

We can merge #8 to have linting running automatically for PRs

File: README.md [hunk 1]
```diff
@@ -133,12 +131,12 @@ Data viewers/Editors based on shapes.
...
 - ShEx
-  - [Shape Expressions Language 2.1](https://shex.io/shex-semantics/index.html) - Final Community Group Report 8 October 2019.
-
-- ShEx, Drafts
-  - [P3330TM/D3 Draft Recommended Practice for Standard for Shape Expression Schemas](https://shexspec.github.io/spec/)
+  - [Shape Expressions Language 2.1](https://shex.io/shex-semantics/index.html) - Final Community Group Report: 8 October 2019
+  - [ShEx Primer](https://shexspec.github.io/primer/index.html) - Draft Community Group Report: 7 December 2022
+  - [ShEx Text Suite](https://github.com/shexSpec/shexTest) and [website](https://shexspec.github.io/shexTest/) - Version `next@1.0.2`: last release 5 October 2018
+  - [Recommended Practice for Standard for Shape Expression Schemas](https://shexspec.github.io/spec/) - IEEE Computer Society P3330/D3: Draft 9 October 2024. Next-version spec with `EXTENDS` keyword
+  - [ShEx Primer](https://shex.io/primer-next/) - Primer with `EXTENDS` keyword: Draft Community Group Report 25 May 2022
```
Comment (TallTed) [inline README.md:138]: These colons are inappropriate for these placements. I don't care if programmers dislike commas. They're sometimes the appropriate punctuation.

**Comment by amivanoff** [comment]
I clarified the item grammar a bit (Description must begin with a capital letter and last DOT).

**1) A simple item without a description**
```SPACE* DASH SPACE <name with hyperlink, images and bages>```

Example
```- [SHACL To JSON Schema](https://github.com/comake/shacl-to-json-schema)```

**2) An item with a description**
```SPACE* DASH SPACE <name with hyperlink, images and bages> SPACE DASH SPACE <Description must begin with a capital letter> DOT```

Example
```- [SHACL To JSON Schema](https://github.com/comake/shacl-to-json-schema) NPM Version GitHub last commit - SHACL-to-JSON-Schema translator; `Typescript`.```

**Linting status**

There are some errors:
The error in line 43 is about first capital letter in a description.
The rest errors are about missing final DOT.

```
  ✖   43:58   List item description must start with valid casing           remark-lint:awesome-list-item
  ✖  100:63   List item description must end with proper punctuation       remark-lint:awesome-list-item
  ✖  104:102  List item description must end with proper punctuation       remark-lint:awesome-list-item
  ✖  134:80   List item description must end with proper punctuation       remark-lint:awesome-list-item
  ✖  135:64   List item description must end with proper punctuation       remark-lint:awesome-list-item
  ✖  137:192  List item description must end with proper punctuation       remark-lint:awesome-list-item
  ✖  138:72   List item description must end with proper punctuation       remark-lint:awesome-list-item
```

It's interesting that this line isn't triggers an error ))
```- [ShEx Text Suite](https://github.com/shexSpec/shexTest) and [website](https://shexspec.github.io/shexTest/) - Version `next@1.0.2`: last release 5 October 2018```

I guess, if there are two links before a description, it skips the final DOT check.

**Comment by TallTed** [comment]
> `SPACE* DASH SPACE`

That should be 

`SPACE* HYPHEN SPACE`

**Comment by VladimirAlexiev** [comment]
fixed all comments in #9, and added `shacl-js`.

**Pull request merged**